test/model/TestCompilerTest.php

```diff
@@ -0,0 +1,140 @@
+     */
+    private $testModel = null;
+
+    public function setUp(){
+        TaoPhpUnitTestRunner::initTest();
+        $this->test = new \core_kernel_classes_Resource('http://myFancyDomain.com/myGreatResourceUriForTest');
+        $this->item = new \core_kernel_classes_Resource('http://myFancyDomain.com/myGreatResourceUriForItem');
+        $this->item->setPropertyValue(new \core_kernel_classes_Property('http://www.tao.lu/Ontologies/TAOItem.rdf#ItemModel'), 'http://www.tao.lu/Ontologies/TAOItem.rdf#QTI');
+        $this->testModel = new TestModel();
+        $this->storage = new \taoDelivery_models_classes_TrackedStorage();
+
+        $this->testModel->save($this->test, array());
+    }
+
+    public function testCompileEmpty() {
+
+        //test without items
+        $testCompiler = new TestCompiler($this->test, $this->storage);
+        $waitingReport = new \common_report_Report(\common_report_Report::TYPE_ERROR, __("A Test must contain at least one item to be compiled."));
+        $report = $testCompiler->compile();
+
+        $this->assertEquals($waitingReport, $report, 'No items in sample test');
+    }
+
+    public function testCompile(){
+        //test with items
+        $this->testModel->save($this->test, array($this->item->getUri()));
+        $waitingReport = new \common_report_Report(\common_report_Report::TYPE_SUCCESS);
+
+        $serviceCall = $this->getMockBuilder('tao_models_classes_service_ServiceCall')
+            ->disableOriginalConstructor()
+            ->setMethods(array('serializeToString'))
+            ->getMock();
+        $serviceCall->expects($this->once())
+            ->method('serializeToString')
+            ->willReturn('greatString');
+
+        $waitingReport->setData($serviceCall);
+
+
+        $testCompiler = $this->getMockBuilder('oat\taoTestLinear\model\TestCompiler')
+            ->setConstructorArgs(array($this->test, $this->storage))
+            ->setMethods(array('subCompile', 'spawnPrivateDirectory'))
+            ->getMock();
+
+        $testCompiler->expects($this->once())
+            ->method('subCompile')
+            ->willReturn($waitingReport);
+
+
+        //will spawn a new directory and store the content file
+        $directoryMock = $this->getMockBuilder('tao_models_classes_service_StorageDirectory')
+            ->disableOriginalConstructor()
+            ->setMethods(array('getPath'))
+            ->getMock();
+
+        $directoryMock->expects($this->once())
+            ->method('getPath')
+            ->willReturn(dirname(__FILE__). '/../sample/compile/');
+
+
+        $testCompiler->expects($this->once())
+            ->method('spawnPrivateDirectory')
+            ->willReturn($directoryMock);
+
+
+
+        $report = $testCompiler->compile();
+
+        $this->assertEquals(__('Test Compilation'), $report->getMessage(),__('Compilation should work'));
+        $this->assertFileExists(dirname(__FILE__). '/../sample/compile/data.json', __('Compilation file not created'));
```

Comment:
folder taoTestLinear/test/sample/compile is not config to be writable for www-data, you should write this file in a /tmp  subfolder using function sys_get_temp_dir()